src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -122,6 +169,50 @@ export class PokemonSaveParser {
     }
   }
 
+  /**
+   * Initialize memory mode with WebSocket client
+   */
+  private async initializeMemoryMode (client: MgbaWebSocketClient): Promise<void> {
+    this.webSocketClient = client
+    this.isMemoryMode = true
+
+    // Check if connected
+    if (!client.isConnected()) {
+      throw new Error('WebSocket client is not connected to mGBA')
+    }
+
+    // Get game title to check compatibility
+    const gameTitle = await client.getGameTitle()
+    if (!this.silent) console.log(`Memory mode: Connected to game "${gameTitle}"`)
+
+    // Auto-detect config based on game title if not provided
+    if (!this.config) {
+      if (gameTitle.includes('EMERALD') || gameTitle.includes('Emerald') || gameTitle.includes('EMER')) {
+        // Use the singleton registry instance
+        const configs = GameConfigRegistry.getRegisteredConfigs()
+        for (const ConfigClass of configs) {
+          try {
+            const config = new ConfigClass()
+            if (config.name.toLowerCase().includes('emerald') || config.name.toLowerCase().includes('vanilla')) {
+              this.config = config
+              break
+            }
+          } catch {
+            continue
+          }
+        }
+
+        if (!this.config) {
+          throw new Error('Emerald config not found in registry')
+        }
+      } else {
+        throw new Error(`Unsupported game for memory parsing: "${gameTitle}". Currently only Pokémon Emerald is supported.`)
+      }
+    }
```

Comment:
the auto detection should be handelet trough config level "canHandle" functions, just like it is with file based parsing.
maybe we should refactor to have a "canHandleFile" and "canHandleMemory" function in the config classes (or whatever is closes to what we actually called the function)